app/services/decline_offer_by_default.rb

```diff
@@ -15,8 +15,8 @@ def call
           ProviderMailer.declined_by_default(provider_user, application_choice).deliver_later
         end
       end
-
-      CandidateMailer.declined_by_default(application_form).deliver_later
     end
+
+    CandidateMailer.declined_by_default(application_form).deliver_later
```

Comment:
:+1: This seems correct! This might be a good time to check we don't have irreversible side-effects inside any other transactions

Reply:
Should we move the `ProviderMailer` invocation out here too?

Reply:
Maybe. I didn't change it earlier because the `ProviderMailer` doesn't do any database queries that depend on the state of the application choices - it's given the application choice id explicitly, whereas the `CandidateMailer` has to find the relevant application choices given only the application form.

Reply:
Sure. Does it still seem a bit odd to make this side-effecting call inside the transaction? ie if we added a line below it which went on to throw an exception, the whole txn would be rolled back but the email would still be sent

Reply:
Thats a good point. Anyway I've had a go at pulling it out of the transaction.